src/components/ha-selector/ha-selector-number.ts

```diff
@@ -51,8 +51,9 @@ export class HaNumberSelector extends LitElement {
             `
           : ""}
         <ha-textfield
-          inputMode="numeric"
-          pattern="[0-9]+([\\.][0-9]+)?"
+          .inputMode=${(this.selector.number.step || 1) < 1
```

Comment:
A step can be `1.5`, making it a decimal.

Reply:
ok, lets just always do decimal then

Reply:
Or:

```suggestion
          .inputMode=${(this.selector.number.step || 1) % 1 != 0
```

Reply:
yeah, but 🤷‍♂️ like the only difference is 1 button on the keyboard on mobile 😄

Reply:
Also in validation right? setting integer will not accept floats in the browser?

Reply:
No, inputmode doesnt do validation, just for what keyboard is shown on a touchscreen

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode:

> It's important to understand that the inputmode attribute doesn't cause any validity requirements to be enforced on input. To require that input conforms to a particular data type, choose an appropriate [<input> element type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types). For specific guidance on choosing [<input>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input) types, see the [Values](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode#values) section.

Reply:
TIL 👍 Thanks!